lib/adal.js

```diff
@@ -1084,17 +1084,21 @@ var AuthenticationContext = (function () {
         // y values are 8, 9, A, B
         var cryptoObj = window.crypto || window.msCrypto; // for IE 11
         if (cryptoObj && cryptoObj.getRandomValues) {
-            var buffer = new Uint16Array(8);
+            var buffer = new Uint8Array(16);
             cryptoObj.getRandomValues(buffer);
+            buffer[6] |= 0x40;
```

Comment:
This looks good to me, please add some comments here explaining how you are setting bits, this will help someone reading the code.

e.g. buffer[6] and buffer[7] represents time_hi_and_version field. We will update the most significant 4 bits of buffer[6] to represent decimal number 4 (uuid version number).
buffer[6] | 01000000 will set the 7 bit to 1
buffer[6] & 01001111 will set the 5, 6, and 8 bit to 0 such that bit 5-8 == 0100 = "4"